async-query/src/main/java/org/opensearch/sql/spark/config/OpenSearchExtraParameterComposer.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.spark.config;
+
+import lombok.RequiredArgsConstructor;
+import org.opensearch.sql.spark.asyncquery.model.AsyncQueryRequestContext;
+import org.opensearch.sql.spark.dispatcher.model.DispatchQueryRequest;
+import org.opensearch.sql.spark.parameter.GeneralSparkParameterComposer;
+import org.opensearch.sql.spark.parameter.SparkSubmitParameters;
+
+/** Load extra parameters from settings and add to Spark submit parameters */
+@RequiredArgsConstructor
+public class OpenSearchExtraParameterComposer implements GeneralSparkParameterComposer {
+  private final SparkExecutionEngineConfigClusterSettingLoader settingLoader;
+
+  @Override
+  public void compose(
+      SparkSubmitParameters sparkSubmitParameters,
+      DispatchQueryRequest dispatchQueryRequest,
+      AsyncQueryRequestContext context) {
+    settingLoader
```

Comment:
do we need Setting loader, why not directly access setting in here.

Reply:
Since it had access from two places, I extracted as a class to avoid redundancy. (It include AccessController.doPrivileged, etc.)